File: scalechain-storage/src/test/scala/io/scalechain/blockchain/storage/index/TransactingRocksDatabaseSpec.scala
package io.scalechain.blockchain.storage.index

import java.io.File

import io.scalechain.blockchain.storage.Storage
import org.apache.commons.io.FileUtils
import org.scalatest.{Suite, BeforeAndAfterEach}

class TransactingRocksDatabaseSpec extends KeyValueDatabaseTestTrait with KeyValueSeekTestTrait with KeyValuePrefixedSeekTestTrait with BeforeAndAfterEach {
  this: Suite =>

  Storage.initialize()

  var db : KeyValueDatabase = null
  var txDb : TransactingRocksDatabase = null


  override def beforeEach() {

    val testPath = new File("./target/unittests-TransactingRocksDatabaseSpec")
    FileUtils.deleteDirectory( testPath )
    txDb = new TransactingRocksDatabase( testPath )
    db = txDb
    txDb.beginTransaction()
    super.beforeEach()
  }

  override def afterEach() {
    super.afterEach()
    txDb.commitTransaction()
    txDb.close()
    db = null
    txDb = null
  }
}

File: scalechain-storage/src/test/kotlin/io/scalechain/blockchain/storage/index/TransactingRocksDatabaseSpec.kt
package io.scalechain.blockchain.storage.index

import io.kotlintest.KTestJUnitRunner
import io.kotlintest.matchers.Matchers
import io.kotlintest.specs.FlatSpec
import java.io.File

import io.scalechain.blockchain.storage.Storage
import org.junit.runner.RunWith

@RunWith(KTestJUnitRunner::class)
class TransactingRocksDatabaseSpec : FlatSpec(), Matchers, DatabaseTestTraits {
  val testPath = File("./build/unittests-TransactingRocksDatabaseSpec")

  lateinit override var db : KeyValueDatabase
  lateinit var txDb : TransactingKeyValueDatabase

  override fun beforeEach() {
    testPath.deleteRecursively()
    testPath.mkdir()

    db = RocksDatabase(testPath)
    txDb = db.transacting()
    txDb.beginTransaction()

    super.beforeEach()
  }

  override fun afterEach() {
    super.afterEach()
    txDb.commitTransaction()
    db.close()
  }

  init {
    Storage.initialize()
    addTests()
  }
}